src/utilities/modules/vaccines.js

```diff
@@ -0,0 +1,41 @@
+import { formattedDifferenceBetweenDates, formatExposureRange } from '../formatters';
+
+/**
+ * Method for creating data for BreachTable rows.
+ * Accepts an array (Example below) of objects,
+ * each having a breach property which contains
+ * a realm results of sensorLog objects.
+ * @param {Array} data Example below
+ * [ {
+ *     breach: [ sensorLogs ],
+ *     ...
+ *   },
+ *    ...
+ * ]
+ */
+export const setBreachData = (data = []) => {
+  const dataCopy = [];
+  data.forEach(dataSet => {
+    const { breach } = dataSet;
+    // A breach must have a minimum of 3 sensorlogs to be considered
+    // a breach.
+    if (!breach || breach.length < 4) return;
+    const [firstSensorLog = null, , { timestamp: lastTimestamp } = {}] = breach;
+    // If there is no first or last sensor logs, or if either does not have a timestamp,
+    // don't create data for this breach.
+    if (!(firstSensorLog && firstSensorLog.timestamp && lastTimestamp)) return;
+    dataCopy.push({
+      ...dataSet,
+      breachData: {
+        affectedQuantity: firstSensorLog.sum('itemBatches.numberOfPacks'),
```

Comment:
should this not come from dataSet.items.batches ?
can modify sensorLogsExtractBatches to also have all batches (ungrouped) in dataSet, that will make this easy

Reply:
I suppose it should. I think I figured it was going to be pretty inaccurate anyway you slice it but that does seem closer and less lazy on my part. Another option is to just calculate some of these statistics in `sensorLogsExtractBreaches` and pass them rather than having the extra field with all the batches?

Reply:
if items.batches are not used from params it would be very inaccurate for a breach for a certain ItemBatch.

sure i am happy if you add the aggregation it to `sensorLogsExtractBreaches` method